src/main/java/org/osiam/addons/self_administration/Config.java

```diff
@@ -118,43 +131,114 @@
     @Value("${org.osiam.mail.server.smtp.starttls.enable:false}")
     private boolean mailServerStartTlsEnabled;
 
-    private String[] allAllowedFields;
+    private HtmlField[] allAllowedFields;
+    private String[] allowedFieldNames;
 
     private boolean confirmPasswordRequired = true;
 
     @Autowired
-    private void createAllAllowedFields(@Value("${org.osiam.html.form.fields:}") String[] allowedFields,
-            @Value("${org.osiam.html.form.extensions:}") String[] extensions) {
-        List<String> collectedFields = new ArrayList<>();
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method createAllAllowedFields() has an NPath complexity of 5280](https://www.codacy.com/app/t-pick/addon-self-administration/file/2113712806/issues/source?bid=3145222&fileBranchId=3145225#l140)